src/gm3/util.js

```diff
@@ -436,3 +436,21 @@ export function metersAreaToUnits(meters, units) {
             return meters;
     }
 }
+
+export function convertLength(length, srcUnits, destUnits) {
+    // US survey feet, miles
+    const equivalentMeters = {
+        'ft': 0.3048,
+        'yd': 0.9144,
+        'mi': 1609.347,
+        'in': 0.0254,
+        'm': 1,
+        'km': 1000,
+        "ch": 20.11684
+    };
+    let convertedLength = length * equivalentMeters[srcUnits] / equivalentMeters[destUnits];
+    if (typeof convertedLength === "number" && convertedLength >= 0) {
```

Comment:
This check should be removed.  Negative distances are a real thing and returning NaN is also valid and should be left up to the calling function to determine how to best handle that.